### DISCUSSION TITLE
/etc/wsl.conf metadata doesn't work like drvfs -o metadata

### DISCUSSION THREAD

**Discussion by b-chu** [microsoft/WSL#6209] [General]
# Environment

```none
Windows build number: Win32NT             10.0.19041.0 Microsoft Windows NT 10.0.19041.0
Your Distribution version: 20.04
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.19.128-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Tue Jun 23 12:58:10 UTC 2020
```

# Steps to reproduce

Content of /etc/wsl.conf
```
[automount]
options = "metadata"
```

Trying to ssh
```
$ ssh user@192.168.1.253
The authenticity of host '192.168.1.253 (192.168.1.253)' can't be established.
ECDSA key fingerprint is SHA256:some-random-stuff.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Failed to add the host to the list of known hosts (/home/user/.ssh/known_hosts).
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@         WARNING: UNPROTECTED PRIVATE KEY FILE!          @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
Permissions 0777 for '/home/user/.ssh/id_rsa' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
Load key "/home/user/.ssh/id_rsa": bad permissions
user@192.168.1.253's password:
```

#  Expected behavior

Following https://superuser.com/questions/1183176/can-i-share-my-ssh-keys-between-wsl-and-windows
`$ sudo umount /mnt/d; sudo mount -t drvfs D: /mnt/d -o metadata`

Trying to ssh
```
$ ssh user@192.168.1.253
The authenticity of host '192.168.1.253 (192.168.1.253)' can't be established.
ECDSA key fingerprint is SHA256:some-random-stuff.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Failed to add the host to the list of known hosts (/home/user/.ssh/known_hosts).
Enter passphrase for key '/home/user/.ssh/id_rsa':
```

# Actual behavior

See Steps to reproduce

# Other

How do I set write permissions for /home/user/.ssh/known_hosts without sudo? As you can see under expected behavior "Failed to add the host to the list of known hosts"

**Comment by therealkenc** [comment]
The permissions of `/home/user/...` is not controlled by `metadata` on `/mnt/d` (the file is not on `/mnt/d/...`).

Probably want something like the following. 
```
$ chmod 700 ~/.ssh
$ chmod 600 ~/.ssh/known_hosts ~/.ssh/id_rsa ~/.ssh/authorized_keys ~/.ssh/config
$ chmod 644 ~/.ssh/id_rsa.pub
```

Cannot answer how your `~/.ssh` ended up with `777` perms in the first instance. If it were me, I'd start over with `ssh-keygen` after moving your existing `~/.ssh` out of the way. If you are playing symlink tricks per the superuser post, guidance would be to not, or failing that, the permissions still need to be correct.

**Comment by b-chu** [comment]
The point of metadata is to not have to set permissions. See below.

https://devblogs.microsoft.com/commandline/chmod-chown-wsl-improvements/
https://devblogs.microsoft.com/commandline/automatically-configuring-wsl/
